DashAI/back/models/hugging_face/vit_transformer.py

```diff
@@ -0,0 +1,164 @@
+    """
+    Pre-trained transformer ViT allowing image classification
+    """
+
+    @classmethod
+    def get_schema(cls):
+        with open("DashAI/back/models/parameters/models_schemas/ViT.json") as f:
+            cls.SCHEMA = json.load(f)
+        return cls.SCHEMA
+
+    def __init__(self, model=None, **kwargs):
+        """
+        Initialize the transformer class by calling the pretrained model and its
+        feature extractor. Include an attribute analogous to sklearn's check_is_fitted
+        to see if it was fine-tuned.
+        """
+        self.model_name = "google/vit-base-patch16-224"
+        self.feature_extractor = ViTFeatureExtractor.from_pretrained(self.model_name)
+        self.model = (
+            model
+            if model is not None
+            else ViTForImageClassification.from_pretrained(self.model_name)
+        )
+        self.fitted = bool(model is not None)
+        self.training_args = kwargs
+
+    def get_preprocess_images(self, input_column: str, output_column: str):
+        """Preprocess images for model input
+
+        Parameters
+        ----------
+        input_column : str
+            name of the column containing the images to be preprocessed
+        output_column : str
+            name of the column containing the output labels for the images
+
+        Returns
+        -------
+        Function
+            a function that preprocesses images and outputs a dictionary
+            containing processed images and corresponding labels.
+        """
+
+        def preprocess_images(examples):
+            inputs = self.feature_extractor(
+                images=examples[input_column], return_tensors="pt", size=224
+            )
+            inputs["labels"] = examples[output_column]
+            return inputs
+
+        return preprocess_images
+
+    def fit(self, dataset: DashAIDataset):
+        """Fine-tuning the pre-trained model
+
+        Parameters
+        ----------
+        dataset : DashAIDataset
+            DashAiDataset with training data
+
+        """
+
+        input_column = dataset.inputs_columns[0]
+        output_column = dataset.outputs_columns[0]
+
+        feature_extractor_func = self.get_preprocess_images(input_column, output_column)
+        dataset = dataset.map(feature_extractor_func, batched=True)
+
+        # Arguments for fine-tuning
+        training_args = TrainingArguments(
+            output_dir="DashAI/back/user_models/temp_checkpoints_vit",
+            save_steps=1,
+            save_total_limit=1,
+            **self.training_args,
+        )
+
+        # The Trainer class is used for fine-tuning the model.
+        trainer = Trainer(
+            model=self.model,
+            args=training_args,
+            train_dataset=dataset,
+        )
+
+        trainer.train()
+        self.fitted = True
+        shutil.rmtree(
+            "DashAI/back/user_models/temp_checkpoints_vit", ignore_errors=True
+        )
+        return
```

Comment:
If we follow the scikit-learn convention, this should return to itself (self).

```suggestion
        return self
```